devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -51,423 +46,136 @@ metadata {
+                            [value: 74, color: "#44b621"],
+                            [value: 84, color: "#f1d801"],
+                            [value: 95, color: "#d04e00"],
+                            [value: 96, color: "#bc2323"]
+                    ]
         }
         valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat") {
             state "battery", label: 'Battery \n${currentValue}%', backgroundColor:"#ffffff"
         }
-        valueTile("zigbeeId", "device.zigbeeId", inactiveLabel: true, decoration: "flat") {
-            state "serial", label:'${currentValue}', backgroundColor:"#ffffff"
-        }
         main "switch"
         details(["switch","refresh","temperature","levelSliderControl","battery"])
     }
 }
 
-/**** PARSE METHODS ****/
+def getPRESSURE_MEASUREMENT_CLUSTER() {0x0403}
+def getMFG_CODE() {0x115B}
+
 def parse(String description) {
     log.debug "description: $description"
-
-    Map map = [:]
-    if (description?.startsWith('catchall:')) {
-        map = parseCatchAllMessage(description)
-    }
-    else if (description?.startsWith('read attr -')) {
-        map = parseReportAttributeMessage(description)
-    }
-    else if (description?.startsWith('temperature: ') || description?.startsWith('humidity: ')) {
-        map = parseCustomMessage(description)
-    }
-    else if (description?.startsWith('on/off: ')) {
-        map = parseOnOffMessage(description)
-    }
-
-    log.debug "Parse returned $map"
-    return map ? createEvent(map) : null
-}
-
-private Map parseCatchAllMessage(String description) {
-    log.debug "parseCatchAllMessage"
-
-    def cluster = zigbee.parse(description)
-    log.debug "cluster: ${cluster}"
-    if (shouldProcessMessage(cluster)) {
-        log.debug "processing message"
-        switch(cluster.clusterId) {
-            case 0x0001:
-                return makeBatteryResult(cluster.data.last())
-                break
-
-            case 0x0402:
-                // temp is last 2 data values. reverse to swap endian
-                String temp = cluster.data[-2..-1].reverse().collect { cluster.hex1(it) }.join()
-                def value = convertTemperatureHex(temp)
-                return makeTemperatureResult(value)
-                break
-
-            case 0x0006:
-                return makeOnOffResult(cluster.data[-1])
-                break
+    def event = zigbee.getEvent(description)
+    if (!event) {
+        Map descMap = zigbee.parseDescriptionAsMap(description)
+        if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrInt == 0x0021) {
+            event = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+        } else if (descMap?.clusterInt == PRESSURE_MANAGEMENT_CLUSTER && descMap.attrInt == 0x0020) {
+            event = getPressureResult(Integer.parseInt(descMap.value, 16))
+        }
+    } else if (event.name == "level" && event.value > 100) {
```

Comment:
Is this possible? I don't think `getEvent` will return a level event with a value > 100

Reply:
Looking at the code in `getEvent` it doesn't look like there's any clamping.

Reply:
https://github.com/PhysicalGraph/appengine-zigbee/blob/master/src/main/groovy/physicalgraph/zigbee/Zigbee.groovy#L836 is what I'm looking at.  Since level is one byte 255 is the max it could be and that should be rounded to 100, but maybe I'm missing something.

Reply:
Ah, yeah, it'll just round to 100. Okay I'll need to work on this.